lib/src/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -0,0 +1,105 @@
+    templateUrl: './breadcrumbs.component.html',
+    styleUrls: ['./breadcrumbs.component.scss']
+})
+export class BreadcrumbsComponent implements OnInit, OnDestroy {
+
+    public breadcrumbs: IBreadcrumb[];
+    public routerSubscription: any;
+    @Input() backURL: string;
+    @Input() backShow: string = 'none';
+    @Input() locationLabel: string = '';
+
+    constructor( private activatedRoute: ActivatedRoute, private router: Router ) {
+        this.breadcrumbs = [];
+    }
+
+    ngOnInit() {
+        const ROUTE_DATA_BREADCRUMB: string = 'breadcrumb';
+        // Add the first breadcrumb for the base page
+        let root: ActivatedRoute = this.activatedRoute.root;
+        this.breadcrumbs = this.getBreadcrumbs(root);
+
+        // subscribe to the NavigationEnd event
+        this.routerSubscription = this.router.events.filter(event => event instanceof NavigationEnd).subscribe(event => {
+            // set breadcrumbs
+            root = this.activatedRoute.root;
+            this.breadcrumbs = this.getBreadcrumbs(root);
+            if ( this.breadcrumbs.length > 1 ) {
+                this.backURL = this.breadcrumbs[ this.breadcrumbs.length - 2 ].url;
+                this.backShow = 'inline';
+                this.locationLabel = '';
+            } else {
+                this.backShow = 'none';
+                this.locationLabel = this.breadcrumbs[ this.breadcrumbs.length - 1 ].label;
+            }
+        });
+    }
+
+    private getBreadcrumbs(route: ActivatedRoute, url: string = '', breadcrumbs: IBreadcrumb[] = []): IBreadcrumb[] {
+        const ROUTE_DATA_BREADCRUMB: string = 'breadcrumb';
+
+        // get the child routes
+        let children: ActivatedRoute[] = route.children;
+
+        // return if there are no more children
+        if (children.length === 0) {
+          return breadcrumbs;
+        }
+
+        // iterate over each children
+        for (let child of children) {
+            // verify primary route
+            if (child.outlet !== PRIMARY_OUTLET) {
+                continue;
+            }
+
+            // verify the custom data property "breadcrumb" is specified on the route
+            if (!child.snapshot.data.hasOwnProperty(ROUTE_DATA_BREADCRUMB)) {
+                return this.getBreadcrumbs(child, url, breadcrumbs);
+            }
+
+            // get the route's URL segment
+            let routeURL: string = child.snapshot.url.map(segment => segment.path).join('/');
+
+            // append route URL to URL
+            url += `/${routeURL}`;
+
+            // add breadcrumb
+            let parent: any = this.activatedRoute.parent;
+            let fullURL: string = '';
+            if (parent !== null ) {
+                fullURL += '/' + parent.snapshot.url.map(segment => segment.path).join('/');
+            }
+            fullURL += `${url}`;
+            let breadcrumb: IBreadcrumb = {
+                label: child.snapshot.data[ROUTE_DATA_BREADCRUMB],
```

Comment:
This is probably something for a second pass, but often times routes would look like `/things/1/edit`. It seems like the data provided to the id portion of the route would need to be generic with this scenario. It might be nice if the breadcrumb could be `Home Page > Thing 1 Summary > Edit Thing 1` rather than `Home Page > Summary > Edit`

Reply:
I like this idea, but I'm probably not the best person to implement it.  Might be better for me to get this first pass merged and let folks who've done more with breadcrumbs in their apps (looking at @petermcclanahan and @benjanderson) build out how we might lookup ids and build them into the breadcrumb label.

Reply:
This would be a requirement for atlas to use these bread crumbs

Reply:
@benjanderson - as you and I discussed, I'll get this PR merged, and then if you can submit an enhancement issues that would be perfect